util/chunk/row_container.go

```diff
@@ -137,6 +139,27 @@ func (c *RowContainer) SpillToDisk() {
 	N := c.m.records.inMemory.NumChunks()
 	c.m.records.inDisk = NewListInDisk(c.m.records.inMemory.FieldTypes())
 	c.m.records.inDisk.diskTracker.AttachTo(c.diskTracker)
+
+	defer func() {
+		r := recover()
+		if r == nil {
+			return
+		}
+
+		err := fmt.Errorf("%v", r)
+		c.m.records.spillError = err
+		buf := make([]byte, 4096)
+		stackSize := runtime.Stack(buf, false)
+		buf = buf[:stackSize]
+		logutil.BgLogger().Error("rowContainer panicked", zap.String("stack", string(buf)), zap.Error(err))
```

Comment:
```
if r := recover(); r != nil {
	c.m.records.spillError = fmt.Errorf("%v", r)
	buf := util.GetStack()
	logutil.BgLogger().Error("SpillToDisk panicked", zap.String("stack", string(buf)), zap.Error(err))
}
```

Reply:
After importing the `util`, an error of `import cycle` appears;
I extracted a function to get stacktrace

```
Running tool: /usr/local/go/bin/go test -timeout 30s -coverprofile=/tmp/vscode-goWV0Gny/go-code-cover github.com/pingcap/tidb/util/chunk -v
  

package github.com/pingcap/tidb/util/chunk
imports github.com/pingcap/tidb/util
imports github.com/pingcap/tidb/kv
imports github.com/pingcap/tidb/util/codec
imports github.com/pingcap/tidb/util/chunk: import cycle not allowed
```